### PR TITLE
Support `else if`

### PR DESCRIPTION
**Pull request by ruudk**
To learn more about Go I thought it would be cool to implement [`else if` support](https://github.com/gobuffalo/plush/issues/29).

What do you think about this approach? 

Codeclimate is failing, but I can fix that later if I know I'm on the right track.

### CODE REVIEW

**Comment by stanislas-m** [comment]
Just a random thought: since plush syntax is mostly inspired by `erb`, how about using the same keyword? (so `elsif` instead of `else if`)

That way, we still can use `erb` syntax highlighting for `plush`.

**Comment by markbates** [comment]
Honestly I never liked that elsif syntax. I would rather else if, personally.

**Comment by ruudk** [comment]
I agree, I don't like the `elsif` because I like to more explicit, and write things out. So either `elseif` or `else if`

I will clean up the PR, thanks for your feedback 🙌

**Comment by ruudk** [comment]
And also the reason for choosing `else if` was that it's the same in Go :)

**Comment by stanislas-m** [comment]
That's a valid point. :)

File: parser/parser.go [hunk 1]
```diff
@@ -473,14 +473,40 @@ func (p *parser) parseIfExpression() ast.Expression {
...
+				return nil
+			}
+
+			if !p.expectPeek(token.LBRACE) {
+				return nil
+			}
+
+			ifElseExp.Block = p.parseBlockStatement()
+
+			expression.ElseIf = append(expression.ElseIf, ifElseExp)
```
[the remark this answers is not in the document]
Comment (ruudk) [reply parser/parser.go]: Decided to not split this to a separate function because it's not that much duplication, and it keeps it clear.

**Comment by ruudk** [comment]
I cleaned up the PR. Woud love to get your feedback again :)

**Comment by markbates** [comment]
Love this! Thank you!

**Pull request merged**